icefoganalytics/elcc-data-management
# ELCC-57: Part 1: Additional Tombstone Fields for Early Learning Centers

Fixes https://yg-hpw.atlassian.net/browse/ELCC-57

# Context

As we discussed in our call yesterday, I’d like to requesting expanding the centre information that we capture in the new ELCC database, so that we can improve the reporting options available to us.

The fields that I’d like to add are:
- Licence holder name licenseHolderName?: string(100)
- Contact name (sometimes this differs, if one licence holder has multiple centres each with a manager) contactName?: string(100)
- physical address (street, community, postal code) - Single Field physicalAddress?: string(250)
- mailing address (PO Box/street, community, postal code) - Single Field mailingAddress?: string(250)
- email 1 (public) email?: string(100)
- email 2 (alternate – internal) altEmail?: string(100)
- phone 1 (public) phoneNumber?: string(20)
- phone 2 (alternate – internal) altPhoneNumber?: string(20)
- fax faxNumber?: string(20)
- vendor ID vendorId?: string(20)
- program type (FN/non-FN)- isFirstNationProgram: boolean (default false)
- inspector name inspectorName?: string
- neighborhood neighborhood?: string
- Whitehorse/Community flag - enum Region (Whitehorse =”Whitehorse”, Communities = “Communities”) - we can expand this later as needed

This is more than I anticipated when we initially discussed the addition, so hopefully it’s not too big of a change at this stage in the game. These additions will significantly increase our reporting abilities and will mean that we can consolidate more of our operational needs into a single platform (currently this information is all being stored in an Access database that is used solely for the creation of reports from the data.

# Implementation

Back-filling the default “region” to whitehorse so that I can enable non-nullability on said column.
Switching name of vendorId column to venderIdentifier so as to avoid confusing the column name with an foreign key.
Break Child Care Centre details into its own component.
Change edit dialog to emit on success, allowing parent to decide on action for increased flexibility.

Leaving for future PR: Split centre creation form apart from center update from

# Screenshots

![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/ef49ac83-700c-4d41-8c37-5d5c4dfaeeb9)
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/d61b9fed-2f2a-44ce-9dfd-b25285b1353f)

# Testing Instructions

1. Boot the app via `dev up`
2. Check that the app complies, and that you can log in at http://localhost:8080.
3. Go to http://localhost:8080/child-care-centres
4. Try to add a center from the button in the top right.
5. Make sure that the page redirects after creation.
6. Click on the "pencil" icon of the Child Care Centre Details panel on the left.
7. Update some centre details and save the record.
8. Check that the page doesn't break after saving.
